### PR TITLE
🔀🌐 [pt_PT] update translations for pt_PT

### PR DESCRIPTION
**Pull request by TackleToWin**
**What kind of update does this PR provide?** *Please check*

- [x] new translations / updated translations / translation fixes
- [ ] a new feature
- [ ] a new module
- [ ] a bugfix for an existing feature / module
- [ ] improvement of style or user experience
- [ ] other: *Please fill out*

**Which language(s) did you add/update translations for?**
*use the xx_XX notation for exact language!*
* pt_PT

**What is included in your update?**
* add/updated translations in:
       -  \docs\pt_PT...
       -  \src\i18n\pt_PT...
       -  \src\i18n\pt_PT.ts
       - \src\modules\{module}\i18n...

**Additional notes**

* files in src\i18n\pt_PT: (needs double check numbers/prices/etc)
       - buildings 100%
       - global 100%
       - schoolings 100%
       - vehicles 100%

* files in \docs\pt_PT:
       - apps 100%
       - appstore 100%
       - bugs 100%
       - contributing 100% 
       - error_report 100%
       - faq 100%
       - metadata 100%
       - other 100%
       - README 100%
       - settings 100%
       - suggestions 100%
       - support 100%
       
* files in \docs\pt_PT\contributing:
              - committing 0%
              - introduction 0%
              - prs 0%
              - translations 0%

* translated modules: (\i18n\ folder only)
       - asyncButtons 100%
       - buildingHover 100%
       - buildingListFilter 100%
       - chatExtras 100%
       - clock 100%
       - creditsextension 100%
       - dailyCreditsSummary 100% (still needs to double check everything)
       - dashboard 100%
       - enhancedPOI 100%
       - enhancedTransportRequests 100%
       - extendedBuilding 100%
       - extendedCallList
       - extendedCallWindow
       - extendedMap
       - generalExtensions
       - heatmap
       - hotkeys
       - lssmaql
       - messageTemplates
       - missionHelper
       - notificationAlert
       - overview
       - redesign
       - releasenotes
       - renameFz
       - schoolingOverview
       - shareAlliancePost
       - statusCounter
       - support
       - telemetry
       - template
       - userid

### CODE REVIEW

**Comment by TackleToWin** [comment]
I don't get what is the error on global.ts, I checked building.ts and everything seems fine

**Comment by jxn-30** [comment]
> I don't get what is the error on global.ts, I checked building.ts and everything seems fine

It is a little tricky since the TypeScript-Errors are not very useful at that point…
There are two issues:
1. If a building does not have extensions, a `extensions: []` is required. That causes an error for buildings `2: Posto PEM/AEM` and `20: Posto PEM/AEM (Pequeno)`
2. There is something wrong with the `Expansão de Patrulhamento` extension on buildings `6: Esquadra PSP/Posto GNR` and `19: Esquadra PSP/Posto GNR (Pequeno)`. You can't have set `isVehicleExtension` to false, the property needs to be removed entirely. Also the 35 as index of required extension seems to be wrong but that doesn't cause an error during build time.

**Comment by TackleToWin** [comment]
In metadata.md there is a section that metions `unique` a few times `Unique User ID`, is it `unique` from `the only one` or is it something else?

**Comment by jxn-30** [comment]
In that context it means that there is no other player with the same ID. It is unique over all players within that version.

**Comment by TackleToWin** [comment]
Gotcha, was starting to wonder if it was something else

**Comment by TackleToWin** [comment]
`time of metadata collection` is it duration or what time it was collected?

**Comment by jxn-30** [comment]
It is "what time" -> always the "now" timestamp of when telemetry is executed.

**Comment by TackleToWin** [comment]
is /docs/en_US/other.md updated?

I'm asking because the information in this file usually takes a bit to update

**Comment by TackleToWin** [comment]
settings.md seems to be outdate, there is this line that doesn't have to be there.

`It is planned to add a possibility to export and import the settings as in `LSSM V.3`.`

**Comment by jxn-30** [comment]
> is /docs/en_US/other.md updated?
> 
> I'm asking because the information in this file usually takes a bit to update in others places

everything in that file that is not autogenerated is up-to-date. It doesn't change very often ^^



> settings.md seems to be outdate, there is this line that doesn't have to be there.
> 
> `It is planned to add a possibility to export and import the settings as in LSSM V.3.`

that's true, seems to be outdated :D
let's change the paragraph to
```md
Using the `Export` and `Import` buttons, you can save your settings to a file and/or share them with your friends. In the near future, we would like to offer the possibility to save settings profile-bound. This means that settings are then no longer bound to a device.
```
Do you want to update the en_US and en_GB file accordingly or should we do that?


btw another thing:
for the docs files, please use `lang: pt-PT` instead of `lang: pt_PT`. Otherwise the matching unfortunately doesn't work correct.

**Comment by TackleToWin** [comment]
> or the docs files, please use `lang: pt-PT` instead of `lang: pt_PT`. Otherwise the matching unfortunately doesn't work correct.

Oh sure, I was using `lang: pt_PT` because US files have `lang: en_US` so I though it was how I was doing

I can change, should I open a new PR or each lang?

**Comment by jxn-30** [comment]
> > or the docs files, please use `lang: pt-PT` instead of `lang: pt_PT`. Otherwise the matching unfortunately doesn't work correct.
> 
> Oh sure, I was using `lang: pt_PT` because US files have `lang: en_US` so I though it supposed to be like that

you're right. I only looked at the Readme.md files and those are using the `en-US`  version. weird :thinking: 
So just use the `pt_PT`, that should be fine. Sorry for the inconvenience


> I can change, should I open a new PR for each lang?
if you want to, yes. But for me it's also fine if you do it in here.

**Comment by TackleToWin** [comment]
> you're right. I only looked at the Readme.md files and those are using the `en-US` version. weird 🤔
> So just use the `pt_PT`, that should be fine. Sorry for the inconvenience

no worries 😄 

> if you want to, yes. But for me it's also fine if you do it in here.

I created a new PR so you can update them right away

**Comment by TackleToWin** [comment]
https://docs.lss-manager.de/en_US/apps.html#lssm-v-3-vs-lssm-v-4

if I'm not wrong the setting `releasenotes` from v3 is already on v4 no? 

this way the text `We want to integrate all the features of V.3 into V.4 in time!` can also be removed?

**Comment by TackleToWin** [comment]
Step 3: Activate on README.md file is also outdated, since now it is a icon instead of LSSM V4 in green

**Comment by deoxis9001** [comment]
> if I'm not wrong the setting `releasenotes` from v3 is already on v4 no?
> 
> this way the text `We want to integrate all the features of V.3 into V.4 in time!` can also be removed?
> 
no, there are still modules that don't work on V4 and are only available on V3.

**Comment by TackleToWin** [comment]
> no, there are still modules that don't work on V4 and are only available on V3.

yeah, I noticed that after sending that :P

**Comment by jxn-30** [comment]
> Step 3: Activate on README.md (en_US) file is also outdated, since now it is a icon instead of LSSM V4 in green

you can set it to be the green badge instead of the icon. In german version, we've clarified that but not on other languages.

**Comment by TackleToWin** [comment]
> you can set it to be the green badge instead of the icon. In german version, we've clarified that but not on other languages.

yes I'm aware, it is just that when you install it shows the icon first, you would have to change later. I think that may confuse people.

**Comment by TackleToWin** [comment]
`Do you want to delete this AAO entry?` What does AAO stands for?

**Comment by deoxis9001** [comment]
it's the regulation system in french is RAI
image

**Comment by jxn-30** [comment]
> `Do you want to delete this AAO entry?` What does AAO stands for?

hmm, where is that text? In english it's called "ARR" (Alarm and response regulations). It's these "fast select" badges one can create. There's an entry in the profile dropdown for them but i don't know how they're called in the portugese version?

**Comment by TackleToWin** [comment]
> hmm, where is that text? In english it's called "ARR" (Alarm and response regulations). It's these "fast select" badges one can create. There's an entry in the profile dropdown for them but i don't know how they're called in the portugese version?

yeah I was suspecting that was the case, `GitHub\LSSM-V.4\src\modules\asyncButtons\i18n\pt_PT.json`

**Comment by deoxis9001** [comment]
> > hmm, where is that text? In english it's called "ARR" (Alarm and response regulations). It's these "fast select" badges one can create. There's an entry in the profile dropdown for them but i don't know how they're called in the portugese version?
> 
> yeah I was suspecting that was the case, `GitHub\LSSM-V.4\src\modules\asyncButtons\i18n\en_US.json`

same for other languages en_GB and en_AU

**Comment by TackleToWin** [comment]
Good day!

What does the `Extended guard` refers to?

```
  },
        upgradeBuilding: {
            regex: /(Station (upgraded( \(from small .*? station\))??|constructed)|Extended guard|Cancel:|Refund Building upgrade|Specialization (built|canceled))/u,
            title: 'Station constructed/extended',
            backgroundColor: '#000080',
            textColor: 'white',
        },
```
`src\modules\dailyCreditsSummary\i18n\en_US.ts`

**Comment by jxn-30** [comment]
It's for when more levels of a building are bought.
Be aware that the `regex` texts in that file must exactly match the texts in game (in the credits summary). If you need help with RegEx, feel free to ask :)

**Comment by TackleToWin** [comment]
> Be aware that the `regex` texts in that file must exactly match the texts in game (in the credits summary). If you need help with RegEx, feel free to ask :)

Thanks for letting me know, I was just translating it XD
Where can I find the information for RegEx?

**Comment by jxn-30** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions There is information and links to a cheatsheet and other usefull websites.
If you want to test if your regex is valid (and matches your needs), I can recommend https://regex101.com/. Be aware to select `EcmaScript (JavaScript)` on the left sidebar before.

**Comment by TackleToWin** [comment]
I will be honest with you, I don't understand shit 😰 

```
export default {
    amount: 'Quantidade',
    charts: {
        income: 'Receita',
        expenses: 'Despesas',
    },
    categories: {
        allianceMission: {
            regex: /^\[Aliança\]/u,
            title: 'Missões da Aliança',
            backgroundColor: '#ff2800',
            textColor: 'white',
        },
        mission: {
            regex: '^%missions%( \\(Sistema de Alarme de Incêndio\\))?( - Falso Alarme)?$',
            title: 'Missões Próprias',
            backgroundColor: '#ff2400',
            textColor: 'white',
        },
        cancelledMissions: {
            regex: '^%missions%( \\(Sistema de Alarme de Incêndio\\))? - [cC]ancelado$',
            title: 'Missões Canceladas',
            backgroundColor: '#7C7978',
            textColor: 'white',
        },
        buyVehicle: {
            regex: /Veículo Comprado/u,
            backgroundColor: '#007fff',
            textColor: 'white',
        },
        dailyLogin: {
            regex: /Recompensa de Login Diário/u,
            backgroundColor: '#8db600',
            textColor: 'black',
        },
        buildings: {
            regex: /(Edifício Construído|Edifício demolido sem reembolso|Edifício Reembolsado$)/u,
            title: 'Edifício Construído/Demolido',
            backgroundColor: '#ed872d',
            textColor: 'black',
        },
        upgradeBuilding: {
            regex: /(Estação (atualizada( \(desde estação .*? pequena\))??|construída)|Guarda extendida|Cancelar:|Atualização de Edifício Reembolsada|Especialização (construído|cancelado))/u,
            title: 'Estação construída/extendida',
            backgroundColor: '#000080',
            textColor: 'white',
        },
        task: {
            regex: /Tarefa Completada («.*?»|".*?")/u,
            title: 'Tarefa Completada',
            backgroundColor: '#bb3385',
            textColor: 'white',
        },
        patients: {
            regex: /(Patiente Transporte|Tratamento de Patiente)/u,
            title: 'Patientes',
            backgroundColor: '#fff600',
            textColor: 'black',
        },
        alliancePatients: {
            regex: /Hospital - Aliança/u,
            backgroundColor: '#ffff99',
            textColor: 'black',
        },
        alliancePrisoners: {
            regex: /Polícia - Aliança/u,
            backgroundColor: '#ffff99',
            textColor: 'black',
        },
        prisoners: {
            regex: /Prisioneiro? Transportado/u,
            title: 'Prisioneiro',
            backgroundColor: '#0bda51',
            textColor: 'black',
        },
        schoolings: {
            regex: /(Aplicação para Treino de Aliança|educação)/u,
            title: 'Educação',
            backgroundColor: '#ff00ff',
            textColor: 'white',
        },
        shitComplexes: {
            regex: /(Edifício anexado à Base de Conjunto de Edifícios|Atualizado Base de Conjunto de Edifícios|Atualizado para Conjunto de Edifícios|Atualização da Base de Conjunto de Edifícios terminada)/u,
            title: 'Conjunto de Edifícios',
            backgroundColor: '#b5651d',
            textColor: 'white',
        },
        fireAlarmSystemCancel: {
            regex: /Falso Alarme/u,
            titel: 'Alarmes Falsos',
            backgroundColor: '#c80815',
            textColor: 'white',
        },
    },
    category: 'Categoria',
    title: 'Sumário',
    total: 'Total',
    others: 'Outros',
    export: {
        export: 'Exportar',
        json: {
            raw: 'cru',
            prettified: 'embelezado',
        },
    },
};

```

**Comment by Cr4zyc4k3** [comment]
What is the problem/What point do you not understand? RegEx (regular expression) is used to write check if a text contains a phrase or not. There are some special chars `\d` will return true on any match from 0-9, etc. It looks super confusing, but it is not hard to understand, only to read and write. And even that will become easier if you do it frequently.

To get started with: Open your Credit summary. Select an item (eg. mission). Put that in the lower box on the website and play around with your translated regex until it works out. There is a lot try&error but you will get better. If you're stuck we'll try to help.

The explanation of the given regex from the websites often helps too. If this is still too confusing: Disassemble it into smaller parts until you understand what is happening.

**Comment by TackleToWin** [comment]
I keep getting this error `/ An unescaped delimiter must be escaped; in most languages with a backslash (\)`

**Comment by TackleToWin** [comment]
What is the 'Target' and 'FMS'?

```
    "vehicle-list": {
        "actions": "",
        "building": "Building",
        "caption": "Name",
        "fms": "FMS",
        "target": "Target",
        "type": "Type"
    },
```

`src\modules\dashboard\i18n\en_US.json`

**Comment by TackleToWin** [comment]
What does FMS stands for?

image

**Comment by Cr4zyc4k3** [comment]
FMS is the abbreviation of Funkmeldestatus, which is the german translation for status code (availible on station, on station, approching, on scene, etc. ) for the dispatch. The target is the building/mission id, where the vehicle drives to. In case the vehicle is on scene/on station it is null.

**Comment by Cr4zyc4k3** [comment]
> I keep getting this error `/ An unescaped delimiter must be escaped; in most languages with a backslash (\)`

If it is the first slash followed by a circonflex (`/^`) you can ignore it. Thats just the short code to start at the beginning of a line

**Comment by TackleToWin** [comment]
> If it is the first slash followed by a circonflex (`/^`) you can ignore it. Thats just the short code to start at the beginning of a line

that's the problema, everywhere there is a / it gives error

image

**Comment by Cr4zyc4k3** [comment]
> > If it is the first slash followed by a circonflex (`/^`) you can ignore it. Thats just the short code to start at the beginning of a line
> 
> that's the problema, everywhere there is a / it gives error
> 
> image

In this case: Follow the help and add a slash. And you copied way too much into the regex field. I recommand you to read a short explantion or watch a youtube video about the basics of regex.

**Comment by TackleToWin** [comment]
> > > If it is the first slash followed by a circonflex (`/^`) you can ignore it. Thats just the short code to start at the beginning of a line
> > 
> > 
> > that's the problema, everywhere there is a / it gives error
> > image
> 
> In this case: Follow the help and add a slash. And you copied way too much into the regex field. I recommand you to read a short explantion or watch a youtube video about the basics of regex.

Will do, thanks. First time working with regex 🤷

**Comment by Cr4zyc4k3** [comment]
Ah, I just saw another issue: The first slash is provided by the side. The last slash and the characters are also provided by the side (here they are currently `/mg` but should be `/u`. Just remove these characters from your input and change the last part by clicking on it that it looks like this
Screenshot_20231008-144838_Fennec (1)
(I'm currently on my phone, but you'll get it)

**Comment by TackleToWin** [comment]
> Ah, I just saw another issue: The first slash is provided by the side. The last slash and the characters are also provided by the side (here they are currently `/mg` but should be `/u`. Just remove these characters from your input and change the last part by clicking on it that it looks like this Screenshot_20231008-144838_Fennec (1) (I'm currently on my phone, but you'll get it)

Didn't even think about that, that solves everything, thanks!

**Comment by TackleToWin** [comment]
Haa What did I do? 😰 😰 😰

**Comment by Cr4zyc4k3** [comment]
You merged a branch, which triggered the CI and the run failed due to an invalid regex

**Comment by TackleToWin** [comment]
> You merged a branch, which triggered the CI and the run failed due to an invalid regex

yeah that branch merge was a mistake, and I knew I should send the file with the regex stuff 😞

**Comment by Cr4zyc4k3** [comment]
> > You merged a branch, which triggered the CI and the run failed due to an invalid regex
> 
> yeah that branch merge was a mistake, and I knew I should send the file with the regex stuff 😞

Do you want to revert this merge? Or what is the problem right now?

**Comment by Cr4zyc4k3** [comment]
You can see the changes [here](https://github.com/TackleToWin/LSSM-V.4/commit/1170deb2f0b27e469dad873fd0b7a8d32717ff37) or if you type `git diff HEAD HEAD^1` in your cli. As far as I'm concerned it did not interfere with you changes an just updates the branch with the latest changes from the dev branch.

If you still want to revert this merge, this is **not** recommended since the changes are pushed and can produce a different history if somebody else already works on them. In this case its really unlikely that that that append, but on branches with more maintainers this could lead to a very serious issue.

So: `git revert 1170deb -m 1` (revert the merge commit and keep the first parent) and `git push --force` (push your changes to the branch and ignore that the remote changes are newer) should do the trick.

File: src/modules/dailyCreditsSummary/i18n/pt_PT.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+export default {
+    amount: 'Quantidade',
+    charts: {
+        income: 'Receita',
+        expenses: 'Despesas',
+    },
+    categories: {
+        allianceMission: {
+            regex: /^\[ALIANÇA]/u,
```
Comment (jxn-30) [inline src/modules/dailyCreditsSummary/i18n/pt_PT.ts:9]: The `]` needs to be escaped too: `\]`

**Comment by TackleToWin** [comment]
Is there a way to find the event missions more easily? Kinda need the numbers for extendedCallWindow module but I don't even know which are event missions and which aren't

**Comment by jxn-30** [comment]
The following code snipped exexuted in console in game (F12 -> Console) will group missions by start & end date. That will help you:
```js
fetch('https://www.leitstellenspiel.de/einsaetze.json').then(res => res.json()).then(m => console.log(m.groupBy(m => `${m.additional.date_start} - ${m.additional.date_end}`)));
```

**Comment by TackleToWin** [comment]
> ```js
> (
> ```

Not working
image

**Comment by Cr4zyc4k3** [comment]
You need to replace the german url with the .pt version of the game

**Comment by TackleToWin** [comment]
> You need to replace the german url with the .pt version of the game

yeah still not working beacuse idk what is `einsaetze.json`

**Comment by jxn-30** [comment]
try
```js
fetch('/einsaetze.json').then(res => res.json()).then(m => {const g = {}; m.forEach(e => {const d = `${e.additional.date_start} - ${e.additional.date_end}`; if(!g[d]) {g[d] = []} g[d].push(e)}); console.log(g)});
```
that should work.

einsaetze.json is a file that contains the information on all missions in game such as ID, requirements, average credits etc.

**Comment by TackleToWin** [comment]
Right, thats what I though, but wasn't working

**Comment by TackleToWin** [comment]
> try
> 
> ```js
> fetch('/einsaetze.json').then(res => res.json()).then(m => {const g = {}; m.forEach(e => {const d = `${e.additional.date_start} - ${e.additional.date_end}`; if(!g[d]) {g[d] = []} g[d].push(e)}); console.log(g)});
> ```
> 
> that should work.
> 
> einsaetze.json is a file that contains the information on all missions in game such as ID, requirements, average credits etc.

that worked, thanks

**Comment by jxn-30** [comment]
@LSS-Manager deploy

**Comment by TackleToWin** [comment]
`src\modules\extendedCallWindow\i18n\pt_PT.json` Here on arrHighlight part, where can I find the vehicles names in blue to change?

**Comment by Cr4zyc4k3** [comment]
Do you mean `arrHover`? `arrHighlight` has no vehicles. Thats more or less simple:

1. Go to https://www.jogo-operador112.com/aaos/ (or click on the ARR Button in your profile)
2. Select an ARR to edit (doesnt matter which one)
3. Execute the following code in the web console (Ctrl + Shift + K on Firefox, Ctrl + Shift + J on Chrome)
```js
document.querySelectorAll("label.integer.optional").forEach(function(entry){
    if(entry.attributes.for.value.startsWith("aao_")){
        console.log('"'+entry.attributes.for.value.split("aao_")[1]+'":"'+entry.innerText+'"')
    }
})
```
This will generate the you need. Just copy & paste it into the file and let your IDE lint it or execute `./build/build.sh --eslint`


PS: You can link to lines like this https://github.com/LSS-Manager/LSSM-V.4/blob/9b4f9a8c52937b6355cd3205dfd43c11bd2af731/src/modules/extendedCallWindow/i18n/en_US.json#L15 That makes it easier to follow your text :smile:

**Comment by jxn-30** [comment]
What is the current state of this PR @TackleToWin @Cr4zyc4k3 ?

**Comment by Cr4zyc4k3** [comment]
Nothing new from my side

**Comment by Cr4zyc4k3** [comment]
@TackleToWin Anything new here?

**Comment by Cr4zyc4k3** [comment]
@jxn-30 Since there is no sign of life from @TackleToWin : Shall we merge it?

**Review by jxn-30 — APPROVED** [review]

**Pull request merged**